py/nlr.h

```diff
@@ -30,7 +30,6 @@
 // exception handling, basically a stack of setjmp/longjmp buffers
 
 #include <limits.h>
-#include <assert.h>
```

Comment:
I guess it's not strictly needed because assert is only used in a macro here, but now users of this file should get the dependency, but then suppose the macro is swapped for a function it would need to be changed again etc. Might be better to have headers being self-contained, not sure.

Reply:
yes, but as far as I can tell the macros that use assert are only defined when
`MICROPY_DEBUG_NLR` is defined, and in that case assert.h is included.